node/narwhal/src/helpers/ready.rs

```diff
@@ -199,4 +199,28 @@ mod tests {
                 .collect::<IndexMap<_, _>>()
         );
     }
+
+    #[test]
+    fn test_ready_duplicate() {
+        let rng = &mut TestRng::default();
+
+        // Sample random fake bytes.
+        let data = |rng: &mut TestRng| Data::Buffer(Bytes::from((0..512).map(|_| rng.gen::<u8>()).collect::<Vec<_>>()));
```

Comment:
nit: it would likely be more efficient to use [Fill](https://docs.rs/rand/latest/rand/trait.Fill.html) to generate many bytes at once

Reply:
done